pkg/tsdb/prometheus/query_parser.go

```diff
@@ -78,7 +71,7 @@ func createStep(dsInfo *DatasourceInfo, model *QueryModel, query backend.DataQue
 		return time.Duration(0), err
 	}
 
-	if hasQueryInterval {
+	if hasQueryInterval && model.StepMode != "" {
```

Comment:
Did the changes to add `model.StepMS` affected this piece as well, or is it something we missed before?